src/System.Text.Json/src/System/Text/Json/JsonCommentHandling.cs

```diff
@@ -9,21 +9,24 @@ namespace System.Text.Json
     /// </summary>
     public enum JsonCommentHandling : byte
     {
+        // Do not re-order.
+        // We rely on the ordering to quickly check things like IsCommentHandlingSupported
```

Comment:
By "re-order" you mean changing the numbers associated with the names?  That'd be a breaking API change regardless of how we depend on it in implementation.